packages/blockchain/src/index.ts

```diff
@@ -641,6 +646,186 @@ export default class Blockchain implements BlockchainInterface {
     })
   }
 
+  /**
+   * Validates a block header, throwing if invalid. It is being validated against the reported `parentHash`.
+   * It verifies the current block against the `parentHash`:
+   * - The `parentHash` is part of the blockchain (it is a valid header)
+   * - Current block number is parent block number + 1
+   * - Current block has a strictly higher timestamp
+   * - Additional PoW checks ->
+   *   - Current block has valid difficulty and gas limit
+   *   - In case that the header is an uncle header, it should not be too old or young in the chain.
+   * - Additional PoA clique checks ->
+   *   - Checks on coinbase and mixHash
+   *   - Current block has a timestamp diff greater or equal to PERIOD
+   *   - Current block has difficulty correctly marked as INTURN or NOTURN
+   * @param header - header to be validated
+   * @param height - If this is an uncle header, this is the height of the block that is including it
+   */
+  public async validateHeader(header: BlockHeader, height?: bigint) {
+    if (header.isGenesis()) {
+      return
+    }
+    const parentHeader = (await this.getBlock(header.parentHash)).header
+    if (!parentHeader) {
+      const msg = header._errorMsg('could not find parent header')
+      throw new Error(msg)
+    }
+
+    const { number } = header
+    if (number !== parentHeader.number + BigInt(1)) {
+      const msg = header._errorMsg('invalid number')
+      throw new Error(msg)
+    }
+
+    if (header.timestamp <= parentHeader.timestamp) {
+      const msg = header._errorMsg('invalid timestamp')
+      throw new Error(msg)
+    }
+
+    if (this._common.consensusAlgorithm() === ConsensusAlgorithm.Clique) {
+      const period = (this._common.consensusConfig() as CliqueConfig).period
+      // Timestamp diff between blocks is lower than PERIOD (clique)
+      if (parentHeader.timestamp + BigInt(period) > header.timestamp) {
+        const msg = header._errorMsg('invalid timestamp diff (lower than period)')
+        throw new Error(msg)
+      }
+      // Validate clique difficulty
+      await this.consensus.validateDifficulty(header)
+    }
+
+    if (header._common.consensusType() === 'pow') {
+      await this.consensus.validateDifficulty(header)
+    }
```

Comment:
I guess we can (and should) remove this PoW check here, to make sure this works in a generic context.

Reply:
This one is super tricky I've discovered.  Because `this.consensus` is the blockchain's current consensus mechanism, this fails on merge transition blocks where the header is now PoS but the blockchain is still PoW (since the blockchain hasn't switched to PoS so `blockchain.consensus` is `ethashConsensus`.  We should skip this check on the merge block because the PoS difficulty check (i.e. PoS blocks must always have difficulty 0) is done in the block constructor via `block._consensusValidation) and it will always fail if you pass in a new header that's a PoS block but the current chain is either still PoA or PoW.  I'll check for PoS instead and skip if so.